source/local/local.go

```diff
@@ -144,17 +149,37 @@ func (s *localSource) GetFeatures() *nfdv1alpha1.Features {
 	return s.features
 }
 
+func parseComments(lines [][]byte, fileName string) map[string]interface{} {
+	comments := make(map[string]interface{})
+	for _, l := range lines {
+		line := strings.TrimSpace(string(l))
+		if len(line) > 0 && strings.HasPrefix(line, "#") {
+			lineSplit := strings.SplitN(line, "=", 2)
+
+			key := lineSplit[0]
+			value := lineSplit[1]
+			switch key {
+			case ExpiryDateKey:
+				expiryDate, err := time.Parse(time.RFC3339, strings.TrimSpace(value))
+				if err != nil {
+					klog.ErrorS(err, "failed to parse feature file expiry date", "fileName", fileName)
+					continue
+				}
+				comments[ExpiryDateKey] = expiryDate
+			default:
+				klog.ErrorS(fmt.Errorf("unknown comment encountered while parsing features' comments"), "unknown comment", "commentKey", key)
```

Comment:
We don't want to log all comment lines as errors (like `# Some regular comment` should just be skipped). Only if it's `# +some-key ...` we consider it as a tag/directive and log an error if we don't recognize the `some-key`